examples/decoding/cohy_decomposition.py

```diff
@@ -0,0 +1,557 @@
+# The goal of the decoding module approach is to use the information from the first 30
+# epochs to fit the filters, and then use these filters to extract the same components
+# from the last 30 epochs.
+#
+# For this, we instantiate the
+# :class:`~mne_connectivity.decoding.CoherencyDecomposition` class with: the
+# information about the data being fit/transformed (using an :class:`~mne.Info` object);
+# the type of connectivity we want to decompose (here CaCoh); the frequency band of the
+# components we want to decompose (here 15-20 Hz); and the channel indices of the seeds
+# and targets.
+#
+# Next, we call the :meth:`~mne_connectivity.decoding.CoherencyDecomposition.fit`
+# method, passing in the first 30 epochs of data we want to fit the filters to. Once the
+# filters are fit, we can apply them to the last 30 epochs using the
+# :meth:`~mne_connectivity.decoding.CoherencyDecomposition.transform` method.
+#
+# The transformed data has shape ``(epochs x components*2 x times)``, where the new
+# 'channels' are organised as the seed components, then target components. For
+# convenience, the
+# :meth:`~mne_connectivity.decoding.CoherencyDecomposition.get_transformed_indices`
+# method can be used to get the ``indices`` of the transformed data for use in the
+# ``spectral_connectivity_...()`` functions.
+
+# %%
+
+# Fit filters to first 30 epochs
+cacoh = CoherencyDecomposition(
+    info=epochs.info,
+    method="cacoh",
+    indices=indices,
+    mode="multitaper",
+    fmin=FMIN,
+    fmax=FMAX,
+    rank=(3, 3),
+)
+cacoh.fit(epochs[: N_EPOCHS // 2].get_data())
+
+# Use filters to transform data from last 30 epochs
+epochs_transformed = cacoh.transform(epochs[N_EPOCHS // 2 :].get_data())
+indices_transformed = cacoh.get_transformed_indices()
+
+########################################################################################
+# We can now visualise the connectivity in the last 30 epochs of the transformed data,
+# which for reference we will compare to connectivity in the last 30 epochs using
+# filters fit to the data itself, as well as bivariate coherence to again demonstrate
+# the signal-to-noise enhancements the multivariate approach offers.
+#
+# To compute connectivity of the transformed data, it is simply a case of passing to the
+# ``spectral_connectivity_...()`` functions: the transformed data; the indices
+# returned from
+# :meth:`~mne_connectivity.decoding.CoherencyDecomposition.get_transformed_indices`; and
+# the corresponding bivariate method (``"coh"`` and ``"cohy"`` for CaCoh; ``"imcoh"``
+# for MIC).
+#
+# As you can see, the connectivity profile of the transformed data using filters fit on
+# the first 30 epochs is very similar to the connectivity profile when using filters fit
+# on the last 30 epochs. This shows that the filters are generalisable, able to extract
+# the same components of connectivity which they were trained on from new data.
```

Comment:
is it worth pointing out that connectivity estimates *outside* the 15-20 Hz band are *lower* when the filters are trained on different data (i.e., better SNR)?

Reply:
I now see that @wmvanvliet had a similar comment to mine, in a resolved comment below. I do think though that it's worth mentioning the SNR point here too, in addition to the overfitting point made later using the sliding window example.

Reply:
Sure, I can add this. How about something like:

"As you can see, the connectivity profile of the transformed data using filters fit on the first 30 epochs is very similar to the connectivity profile when using filters fit on the last 30 epochs. This shows that the filters are generalisable, able to extract the same components of connectivity which they were trained on from new data. Additionally, by fitting the filters to only a select frequency band, we can avoid overfitting in frequency bins where no interactions are present (here outside of the 15-20 Hz band) and thereby enhance the overall signal-to-noise ratio of our connectivity estimates."